system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.cpp

```diff
@@ -104,4 +108,13 @@ void PeriodicMeasurementNode::performPeriodicMeasurement()
   RCLCPP_DEBUG(this->get_logger(), getStatusString());
 }
 
+MetricsMessage PeriodicMeasurementNode::newMetricsMessage()
```

Comment:
```suggestion
MetricsMessage PeriodicMeasurementNode::newMetricsMessage() const
```

Reply:
+ send a small PR to make now const

Reply:
Relevant PR to `rclcpp` is at https://github.com/ros2/rclcpp/pull/922.

Reply:
I see, it's sad that it's not const yet, but I don't think we necessarily need to overthink it too much in this case, you can let this non-const in this case.

Reply:
The relevant change to `rclcpp` seems to have been merged. I and other developers and will need to update our development environment. Does the CI always build off of the latest ROS2 master?